pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,269 @@
+					},
+					// PasswordHash: &pwdHash,
+				},
+			},
+		},
+	}
+
+	// a.RendezvousIP = nodeZeroIP
+	// Default to x86_64
+	archName := arch.RpmArch(types.ArchitectureAMD64)
+	if infraEnv.Spec.CpuArchitecture != "" {
+		archName = infraEnv.Spec.CpuArchitecture
+	}
+	a.archName = archName
+
+	releaseImageList, err := releaseImageList(clusterImageSet.Spec.ReleaseImage, archName)
+	if err != nil {
+		return err
+	}
+
+	registriesConfig := &mirror.RegistriesConf{}
+	// registryCABundle := &mirror.CaBundle{}
+	// dependencies.Get(registriesConfig, registryCABundle)
+
+	publicContainerRegistries := getPublicContainerRegistries(registriesConfig)
+
+	logrus.Infof("The release image is %s", clusterImageSet.Spec.ReleaseImage)
+	releaseImageMirror := mirror.GetMirrorFromRelease(clusterImageSet.Spec.ReleaseImage, registriesConfig)
+
+	infraEnvID := uuid.New().String()
+	logrus.Debug("Generated random infra-env id ", infraEnvID)
+	a.infraEnvID = infraEnvID
+
+	osImage, err := getOSImagesInfo(archName)
+	if err != nil {
+		return err
+	}
+	a.osImage = *osImage
+	a.CPUArch = *osImage.CPUArchitecture
+
+	agentTemplateData := &agentTemplateData{
+		PullSecret:                pullSecretAsset.GetPullSecretData(),
+		ReleaseImages:             releaseImageList,
+		ReleaseImage:              clusterImageSet.Spec.ReleaseImage,
+		ReleaseImageMirror:        releaseImageMirror,
+		HaveMirrorConfig:          len(registriesConfig.MirrorConfig) > 0,
+		PublicContainerRegistries: publicContainerRegistries,
+		InfraEnvID:                a.infraEnvID,
+		OSImage:                   a.osImage,
+		Proxy:                     infraEnv.Spec.Proxy,
+	}
+
+	err = bootstrap.AddStorageFiles(&config, "/", "agent/files", agentTemplateData)
+	if err != nil {
+		return err
+	}
+
+	// Set up bootstrap service recording
+	if err := bootstrap.AddStorageFiles(&config,
+		"/usr/local/bin/bootstrap-service-record.sh",
+		"bootstrap/files/usr/local/bin/bootstrap-service-record.sh",
+		nil); err != nil {
+		return err
+	}
+
+	// Use bootstrap script to get container images
+	relImgData := struct{ ReleaseImage string }{
+		ReleaseImage: clusterImageSet.Spec.ReleaseImage,
+	}
+	for _, script := range []string{"release-image.sh", "release-image-download.sh"} {
+		if err := bootstrap.AddStorageFiles(&config,
+			"/usr/local/bin/"+script,
+			"bootstrap/files/usr/local/bin/"+script+".template",
+			relImgData); err != nil {
+			return err
+		}
+	}
+
+	filesToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
+	}
+
+	for _, file := range filesToInclude {
+		manifestFile := ignition.FileFromBytes(filepath.Join(manifestPath, filepath.Base(file.Filename)),
+			"root", 0600, file.Data)
+		config.Storage.Files = append(config.Storage.Files, manifestFile)
+	}
+
+	disabledServices := []string{"set-hostname.service", "start-cluster-installation.service"}
+	enabledServices := StringArrayRemove(agentEnabledServices, disabledServices)
```

Comment:
It's strange to see services being removed from the list in ignitionbase rather than added in the fully configured ignition.

Reply:
Ok, I'll switch it to having a default list in IgnitionBaseAsset and then in the IgnitionAsset add in the additional services.